### PR TITLE
Optionalize app.run dictConfig (fix #1000)

### LINKED ISSUE #1000: Do not override log configuration
Hello.

Can you optionalize the `logging.config.dictConfig` in `app.py`? My application already have a comprehensive log configuration (and does not use log conf), I don't need to change it.

For the moment, my workaround is to override the `Sanic` class:
```

class UseMyLoggingSanic(Sanic):

    def __init__(self, name=None, router=None, error_handler=None,
                 load_env=True, request_class=None,
                 strict_slashes=False):

        from collections import deque, defaultdict
        from inspect import stack, getmodulename
        from sanic.router import Router
        from sanic.handlers import ErrorHandler
        from sanic.config import Config

        # Get name from previous stack frame
        if name is None:
            frame_records = stack()[1]
            name = getmodulename(frame_records[1])

        # logging
        # logging.config.dictConfig(log_config or LOGGING_CONFIG_DEFAULTS)
        self.log_config = None

        self.name = name
        self.router = router or Router()
        self.request_class = request_class
        self.error_handler = error_handler or ErrorHandler()
        self.config = Config(load_env=load_env)
        self.request_middleware = deque()
        self.response_middleware = deque()
        self.blueprints = {}
        self._blueprint_order = []
        self.debug = None
        self.sock = None
        self.strict_slashes = strict_slashes
        self.listeners = defaultdict(list)
        self.is_running = False
        self.is_request_stream = False
        self.websocket_enabled = False
        self.websocket_tasks = set()

        # Register alternative method names
        self.go_fast = self.run
```

This way, logs are routed correctly in my debug terminal, and not twice

### PR DESCRIPTION
**Pull request by Stibbons**
Signed-off-by: Gaetan Semet <gaetan@xeberon.net>

### CODE REVIEW

**Comment by joar** [comment]
Could you also make

https://github.com/Stibbons/sanic/blob/b6fcbcbde3e520220758d515de329ca2d1da12be/sanic/app.py#L797-L800

optional?

**Comment by Stibbons** [comment]
@joar is it what you wanted?

File: sanic/app.py [hunk 1]
```diff
@@ -28,15 +28,16 @@ class Sanic:
 
     def __init__(self, name=None, router=None, error_handler=None,
                  load_env=True, request_class=None,
-                 strict_slashes=False, log_config=None):
+                 strict_slashes=False, log_config=None, configure_logging=True):
```
Comment (r0fls) [inline sanic/app.py]: Add a newline here: `sanic/app.py:31:80: E501 line too long (80 > 79 characters)`

File: sanic/app.py [hunk 2]
```diff
@@ -28,15 +28,17 @@ class Sanic:
 
     def __init__(self, name=None, router=None, error_handler=None,
                  load_env=True, request_class=None,
-                 strict_slashes=False, log_config=None):
+                 strict_slashes=False, log_config=None,
```
Comment (r0fls) [inline sanic/app.py]: remove the extra space here: `strict_slashes=False, log_config=None, ` 

travis message: `sanic/app.py:31:56: W291 trailing whitespace`

**Comment by Stibbons** [comment]
+1

**Pull request merged**